File: src/page/content/todo/material/template/todo.py
from cs50 import get_int, get_string


def show():
    # TODO: Implement this function that lists all todos
    raise NotImplementedError


def create(task):
    # TODO: Implement this function that creates a todo
    raise NotImplementedError


def complete(index):
    # TODO: Implement this function that completes the todo at the given index
    raise NotImplementedError


def delete(index):
    # TODO: Implement this function that deletes the todo at the given index
    raise NotImplementedError


def update(index, new_task):
    # TODO: Implement this function that updates the todo at the given index with the given new_task
    raise NotImplementedError


def main():
    """
    Main function that interacts with the user and calls the other function
    """

    # Welcome the user
    print("Welcome to the ToDo application!")
    print()

    # Repeat until the user aborts
    while True:
        # Show all possible options to the user
        print("1) List ToDos")
        print("2) Create ToDo")
        print("3) Complete ToDo")
        print("4) Delete ToDo")
        print("5) Update ToDo")
        print("0) Exit")

        # Ask the user for an option
        option = get_int("Select an option: ")

        # Act according to the option
        if option == 0:
            print("Goodbye!")
            break
        elif option == 1:
            show()

        elif option == 2:
            # Ask the user for the todos task
            task = get_string("Enter the task: ")
            create(task)

        elif option == 3:
            # Ask the user for the todos index
            index = get_int("Enter the task index: ")

            try:
                complete(index)
            except ValueError as e:
                print(f"INVALID: {str(e)}")

        elif option == 4:
            # Ask the user for the todos index
            index = get_int("Enter the task index: ")

            try:
                delete(index)
            except ValueError as e:
                print(f"INVALID: {str(e)}")

        elif option == 5:
            # Ask the user for the todos index and task
            index = get_int("Enter the task index: ")
            new_task = get_string("Enter the updated task: ")

            try:
                update(index, new_task)
            except ValueError as e:
                print(f"INVALID: {str(e)}")
        else:
            # Tell the user that his selection was wrong
            print("ERROR: Invalid selection, try again!")


if __name__ == "__main__":
    main()

File: src/page/content/todo/material/template/todo.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>

void show(void);
void create(string task);
void complete(int index);
void delete(int index);
void update(int index, string new_task);

// Main function that interacts with the user and calls the other function
int main(void)
{
    // Welcome the user
    printf("Welcome to the ToDo application!\n");
    printf("\n");

    // Repeat until the user aborts
    while (true)
    {
        // Show all possible options to the user
        printf("1) List ToDos\n");
        printf("2) Create ToDo\n");
        printf("3) Complete ToDo\n");
        printf("4) Delete ToDo\n");
        printf("5) Update ToDo\n");
        printf("0) Exit\n");

        // Ask the user for an option
        int option = get_int("Select an option: ");

        // Act according to the option
        if (option == 0)
        {
            printf("Goodbye!\n");
            break;
        }
        else if (option == 1)
        {
            show();
        }
        else if (option == 2)
        {
            // Ask the user for the todos task
            string task = get_string("Enter the task: ");
            create(task);
        }
        else if (option == 3)
        {
            // Ask the user for the todos index
            int index = get_int("Enter the task index: ");
            complete(index);
        }
        else if (option == 4)
        {
            // Ask the user for the todos index
            int index = get_int("Enter the task index: ");
            delete (index);
        }
        else if (option == 5)
        {
            // Ask the user for the todos index and task
            int index = get_int("Enter the task index: ");
            string new_task = get_string("Enter the updated task: ");

            update(index, new_task);
        }
        else
        {
            // Tell the user that his selection was wrong
            printf("ERROR: Invalid selection, try again!\n");
        }
    }

    return 0;
}

void show(void)
{
    // TODO: Implement this function that lists all todos
    printf("ERROR: Not implemented, yet!\n");
}

void create(string task)
{
    // TODO: Implement this function that creates a todo
    printf("ERROR: Not implemented, yet!\n");
}

void complete(int index)
{
    // TODO: Implement this function that completes the todo at the given index
    printf("ERROR: Not implemented, yet!\n");
}

void delete(int index)
{
    // TODO: Implement this function that deletes the todo at the given index
    printf("ERROR: Not implemented, yet!\n");
}

void update(int index, string new_task)
{
    // TODO: Implement this function that updates the todo at the given index with the given new_task
    printf("ERROR: Not implemented, yet!\n");
}
